nuxeo-core/nuxeo-core-schema/src/main/java/org/nuxeo/ecm/core/schema/SchemaManagerImpl.java

```diff
@@ -237,6 +237,22 @@ public DocumentTypeDescriptor getDocumentTypeDescriptor(String name) {
         return last;
     }
 
+    // NXP-14218: used for tests, to be able to unregister it
+    public FacetDescriptor getFacetDescriptor(String name) {
+        FacetDescriptor desc = allFacets.stream().filter(f -> f.name.equals(name)).reduce((a, b) -> b).get();
```

Comment:
Here you could probably replace `reduce((a, b) -> b).get()` by `findFirst().get()`or `findAny().get()`.
As it, we won't compute all the stream to get one value, with `find*` methods, stream is closed since it gets a value.